xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-test/xwiki-platform-ckeditor-test-pageobjects/src/main/java/org/xwiki/ckeditor/test/po/image/ImageDialogSelectModal.java

```diff
@@ -57,7 +57,7 @@ private ImageDialogSelectModal(By selector)
      */
     public ImageDialogEditModal clickSelect()
     {
-        WebElement element = this.container.findElement(By.className("btn-primary"));
+        WebElement element = this.container.findElement(By.cssSelector(".modal-footer .btn-primary"));
```

Comment:
This is good for bulletproofing, but I think it shows we have an issue at UIX level: there should be ideally only a single primary button (action) on the modal.

Reply:
This is not just bulletproofing, the dependencies I add make the "Gallery" tab appear in the image selection modal. This tab contains a primary button which is first in the DOM Tree. (To select wether we want to look for images from the current page or from the whole instance)
I needed to add that selector to make sure we find the select button.
Should I open an issue for these buttons?

Reply:
> the dependencies I add make the "Gallery" tab appear in the image selection modal. This tab contains a primary button which is first in the DOM Tree

Yes, that's what I mean (making the selector more specific to be sure we target the right button). But the fact that we need to do this shows that there can be multiple primary buttons on the Select Image modal, which is not ideal for UIX. I think it's worth reporting an issue. I'm interested to know why @manuelleduc made those buttons from the Gallery tab primary. It may be for a good reason.